### ISSUE TITLE
Permission-Check in tl_news uses wrong ID for picker when no admin user

### ISSUE THREAD

**Issue by koertho** [contao/contao#3011]
**Affected version(s)**

Contao 4.9

**Description**

When a non-admin user uses the picker for tl_news, the tl_news::checkPermission()-Method use the wrong ID for checking permission. This problem only occurs if the source entity (where the picker field is locatet, for example tl_article) has an id, that don't exist in tl_news (for example there is an article with id 4 but no news with id 4).

Steps to reproduce (use tl_article as example):
1. Create a picker input in the dca
2. Login as a backend user without admin rights
3. Edit an article with id 2 when there is no news with id 2
4. Choose a news entry with the picker - it stay empty

The behavior is triggered by this line https://github.com/contao/contao/blob/204fb5a978e1cd49bc2819bdc34df421a1cd1296/news-bundle/src/Resources/contao/dca/tl_news.php#L548 (where the id is read from get parameter, but the get parameter is the id of the source entity, tl_article in my example) and afterwards it throws the exeption her: https://github.com/contao/contao/blob/204fb5a978e1cd49bc2819bdc34df421a1cd1296/news-bundle/src/Resources/contao/dca/tl_news.php#L604

Thread on slack: https://contao.slack.com/archives/CK4J0KNDB/p1621256386378300

**Comment by rabauss** [comment]
If you want to use other entities in a picker I guess this always a problem for all of them (events, faq, ...) with similar `checkPermission` with the line:
```php
$id = strlen(Input::get('id')) ? Input::get('id') : CURRENT_ID;
```

**Comment by koertho** [comment]
I don't think this one should be closed?

**Comment by aschempp** [comment]
I just tried to reproduce this in the online demo, but I had no issue:

- Edit the content element with ID 149
- Set content element type to "include element", I was able to navigate through news, edit news, pick an element from a news
- Set content element type to "hyperlink" and pick a hyperlink URL. I was able to navigate through news, edit news and pick a news

The picker runs in an iframe which does not have an ID in the URL (at least not in my case), so I'm unsure if this issue description is correct (but maybe there's another source).

> (where the id is read from get parameter, but the get parameter is the id of the source entity, tl_article in my example)

**Comment by koertho** [comment]
I just tried with an empty Contao 4.11.5 instance:

Create an dca file:
```
# contao/dca/tl_article.php

use Contao\CoreBundle\DataContainer\PaletteManipulator;

$dca                             = &$GLOBALS['TL_DCA']['tl_article'];
$dca['fields']['someNewsPicker'] = [
    'inputType' => 'picker',
    'relation' => ['type' => 'hasOne', 'load' => 'eager', 'table' => 'tl_news'],
    'sql'                     => "int(10) unsigned NOT NULL default 0",
];
PaletteManipulator::create()
    ->addField('someNewsPicker','inColumn', PaletteManipulator::POSITION_BEFORE)
    ->applyToPalette('default', 'tl_article');
```

In first place I also got an issue with news archive: 
1. Create an article with id 1 (there are no other articles)
2. Create a news archive with id 2 (there is no news archive with id 1)
3. Create a user with access to this news archive and to the article.
4. Switch to the created user
5. Edit the article with id 1 and click the picker:

![image](https://user-images.githubusercontent.com/12064642/127462312-ce5fe516-8b95-4d59-929b-70ff67a99c1c.png)

(to clarify: the news archive has id **2** and the article id 1, but I think this is an issue with the session)

After creating an article with id 2 and some reloading it worked. 

Afterwards we come to the origin reported problem:
1. Create a news entry with id 2 (there is no news with id 1)
2. Open the article with id 1.
3. Select the news with id 2 and click "Anwenden"
4. The picker widget stays empty and in the console you can see the ajax request returns forbidden (403)

We can now open article 2 and select news 2, and it will work.

**Comment by fritzmg** [comment]
I can confirm the issue in the Contao Online Demo:

1. Log in as `j.wilson`.
2. Edit a news article and insert a new content element of type `alias`.
3. Click on _Change selection_ for **Referenced element**.

The following error will appear:

```
Invalid content element ID X.
```

This was a known limitation of the old picker, but should not be with the new picker?

// sorry, got confused, that is a different issue: https://github.com/contao/contao/issues/2046

**Comment by leofeyer** [comment]
@koertho Can you please check whether #4819 fixes your issue?

**Comment by koertho** [comment]
@leofeyer I've tested it today, but it don't fix this issue. The id is retrived from the GET parameter here: 
https://github.com/contao/contao/blob/204fb5a978e1cd49bc2819bdc34df421a1cd1296/news-bundle/src/Resources/contao/dca/tl_news.php#L548

**Comment by koertho** [comment]
Also the CURRENT_ID constant is wrong here: The selected news entry has id 1, the selected article has id 3. Input::get('id') has value 3, CURRENT_ID has value 2.

EDIT:
I also tried changing the id evaluation to `$id = $this->findCurrentId('tl_news');`, but the method return null in this case.

**Comment by leofeyer** [comment]
@ausi /cc

**Comment by ausi** [comment]
I was not able to reproduce the first issue from the reproduction steps in https://github.com/contao/contao/issues/3011#issuecomment-888945467 but I was able to reproduce the second.

Closed in favor of #5143
